### DISCUSSION TITLE
Is there a list of all the font features supported and their meaning?

### DISCUSSION THREAD

**Discussion by ecerulm** [tonsky/FiraCode#1513] [General]
From https://github.com/tonsky/FiraCode/tree/master/features
I can see `calt`, 'cv01`, `cv02`, etc but I don't see a textual explanation of what those features are supposed to do. 

For JetBrains Mono they have this page https://github.com/JetBrains/JetBrainsMono/wiki/OpenType-features where they explain the font features available and what they do.

**Comment by tonsky** [comment]
There’s a picture in the README

![](https://github.com/tonsky/FiraCode/raw/master/extras/ligature_variants.png)

**Reply by ecerulm** [reply]
Yes, I see, since they are in the image I was not able to find (by text search) . 

I created https://github.com/tonsky/FiraCode/pull/1514 to add `calt`, `onum`, `zero`, `cv01`, `ss01` to the README.md so that it's easier to find in the future. 

By the way `calt` in particular I don't think it's explained in the images either, to understand that that does you need to read the files at https://github.com/tonsky/FiraCode/tree/master/features/calt , right? I would be good to add some description of `calt` to the README.md or yet another image that shows what `calt` does.

**Reply by tonsky** [reply]
calt turns everything off, probably not something you would want with Fira Code

**Reply by ecerulm** [reply]
But `+calt` does not turn everything off, does it? I think `calt` is enabled by default,  I tried with `+calt` and `->` is shown as a continuous arrow, when I try with `-calt` there is distinct hyphen and angle bracket. 

If you want I can write in the README.md something like `calt` (contextual alternates) is enabled by default, disabling this feature will remove all ligatures/arrows (not recommended). 

Also I discovered the https://otf.show site where it's easy to see the effect of the standard opentype font features that can be enabled/disabled in Fira Code

* `calt`: https://otf.show/calt?font=Fira%20Code
* `onum`: https://otf.show/onum?font=Fira%20Code
* `frac`: https://otf.show/frac?font=Fira%20Code
* `zero`: https://otf.show/zero?font=Fira%20Code

Do you want me to link to those from the README?

**Reply by tonsky** [reply]
I think we’re fine as it is
